src/cromshell/utilities/http_utils.py

```diff
@@ -61,3 +63,23 @@ def check_http_request_status_code(
                 f"Status_code: {response.status_code}\n"
                 f"Message: {response.text}"
             )
+
+
+def generate_headers(config):
+    """
+    Check the config for options that require a header and generate the appropriate map.
+    Will be an empty map if no relevant options are specified.
+    """
+    headers = {}
+
+    if config.referer_header_url is not None:
+        headers["Referer"] = config.referer_header_url
+
+    if config.gcloud_token_email is not None:
+        stream = os.popen(
+            f"gcloud auth --account={config.gcloud_token_email} print-access-token"
```

Comment:
Wow, I just completely failed to read your comments.  So this should crash with a clear error then?

Reply:
I learned that Python has 23849279 ways to call out to a subprocess and the one I was using was one of the 4989023 considered deprecated.

I upgraded to a cooler version called `check_output()` that grabs stdout on exit success, or raises an error with stderr in case of failure.

Cromshell now halts and prints a specific error like this:
```
subprocess.CalledProcessError: Command '['gcloud', 'auth', '--account=asdf', 'print-access-token']' returned non-zero exit status 1.
[...]
ERROR: (gcloud.auth.print-access-token) Your current active account [asdf] does not have any valid credentials
```
Or like this:
```
FileNotFoundError: [Errno 2] No such file or directory: 'geecloud'
```